libr/anal/fcn.c

```diff
@@ -401,23 +438,21 @@ static RAnalBlock *appendBasicBlock(RAnal *anal, RAnalFunction *fcn, ut64 addr)
 }
 
 #define FITFCNSZ() {\
-		st64 n = bb->addr + bb->size - fcn->addr;\
-		if (n >= 0 && r_anal_fcn_size (fcn) < n) { r_anal_fcn_set_size (NULL, fcn, n); } }\
+	st64 n = bb->addr + bb->size - fcn->addr;\
+	if (n >= 0 && r_anal_fcn_size (fcn) < n) {\
+		r_anal_fcn_set_size (NULL, fcn, n); }\
+	}\
 	if (r_anal_fcn_size (fcn) > MAX_FCN_SIZE) {\
 		/* eprintf ("Function too big at 0x%"PFMT64x" + %d\n", bb->addr, fcn->size); */\
 		r_anal_fcn_set_size (NULL, fcn, 0);\
-		return R_ANAL_RET_ERROR; }
+		return R_ANAL_RET_ERROR;\
+	}
 
-static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut64 len, int depth);
+static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 len, int depth);
 #define recurseAt(x) {\
-		ut8 *bbuf = malloc (anal->opt.bb_max_size);\
-		if (bbuf) {\
-			anal->iob.read_at (anal->iob.io, x, bbuf, anal->opt.bb_max_size);\
-			ret = fcn_recurse (anal, fcn, x, bbuf, anal->opt.bb_max_size, depth - 1);\
-			r_anal_fcn_update_tinyrange_bbs (fcn);\
-			r_anal_fcn_set_size (anal, fcn, r_anal_fcn_size (fcn));\
-			free (bbuf);\
-		}\
+	ret = fcn_recurse (anal, fcn, x, anal->opt.bb_max_size, depth - 1);\
```

Comment:
I guess here you wanted to use `len` instead of `anal->opt.bb_max_size`, right?

Reply:
Afaik the only place that `len` matters i.e. it may not be equal to `anal->opt.bb_max_size`, is in `r_anal_fcn_add_bb()`, which only recurses to a depth of 1.

Reply:
wasnt this about removing bbmaxsize? :D also this is fcnsz not bbsz

Reply:
> wasnt this about removing bbmaxsize? :D

eventually but lets get it green first

> also this is fcnsz not bbsz

?

Reply:
I'm using `fcn_recurse()` here because it updates important vars like `bb->ninstr` and `bb->op_pos`.